src/kotlin_instance.cpp

```diff
@@ -1,5 +1,15 @@
 #include "kotlin_instance.h"
 #include "kotlin_language.h"
+#include "kt_class.h"
+
+KotlinInstance::KotlinInstance(KtClass *ktClass, Object *p_owner) :
+ktObject(ktClass->create_instance(jni::Jvm::current_env(), nullptr, 0, p_owner)) {
```

Comment:
KotlinInstance should accept `jni::JObject` param instead instead of calling create instance here.

KtObject should also store a pointer to a `KtClass` that created it.